drivers/flash/soc_flash_nrf5.c

```diff
@@ -165,8 +201,339 @@ static int nrf5_flash_init(struct device *dev)
+		NULL, /* no op callback */
+		NULL);
+
+	if ((err != TICKER_STATUS_SUCCESS) && (err != TICKER_STATUS_BUSY)) {
+		((struct flash_op_desc *)context)->result = -ECANCELED;
+
+		/* abort flash timeslots */
+		err = ticker_stop(instance_index, 0, ticker_id, NULL, NULL);
+
+		if ((err != TICKER_STATUS_SUCCESS)
+			&& (err != TICKER_STATUS_BUSY)) {
+			__ASSERT(0, "Failed to stop ticker.\n");
+		}
+
+		/* notify thread that data is available */
+		k_sem_give(&sem_sync);
+	}
+}
+
+static int work_in_time_slice(struct flash_op_desc *p_flash_op_desc)
+{
+	u8_t instance_index;
+	u8_t ticker_id;
+	int result;
+	u32_t err;
+
+	ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+	err = ticker_start(instance_index,
+			   3, /* user id for thread mode */
+			      /* (MAYFLY_CALL_ID_PROGRAM) */
+			   ticker_id, /* flash ticker id */
+			   ticker_ticks_now_get(), /* current tick */
+			   0, /* first int. immediately */
+			   TICKER_US_TO_TICKS(FLASH_INTERVAL), /* periodic */
+			   TICKER_REMAINDER(FLASH_INTERVAL), /* per. remaind.*/
+			   0, /* lazy, voluntary skips */
+			   TICKER_US_TO_TICKS(FLASH_SLOT),
+			   time_slot_callback_helper,
+			   p_flash_op_desc,
+			   NULL, /* no op callback */
+			   NULL);
+
+	if ((err != TICKER_STATUS_SUCCESS) && (err != TICKER_STATUS_BUSY)) {
+		result = -ECANCELED;
+	} else if (k_sem_take(&sem_sync, K_MSEC(FLASH_TIMEOUT_MS)) != 0) {
+		/* wait for operation's complete overrun*/
+		result = -ETIMEDOUT;
+	} else {
+		result = p_flash_op_desc->result;
+	}
+
+	return result;
+}
+
+static int erase_in_timeslice(u32_t addr, u32_t size)
+{
+	struct erase_context context = {
+			addr,
+			size,
+			1 /* enable time limit */
+	};
+
+	struct flash_op_desc flash_op_desc = {
+			erase_op,
+			&context
+	};
+
+	return work_in_time_slice(&flash_op_desc);
+}
+
+static int write_in_timeslice(off_t addr, const void *data, size_t len)
+{
+	struct write_context context = {
+			(u32_t) data,
+			addr,
+			len,
+			1 /* enable time limit */
+	};
+
+	struct flash_op_desc flash_op_desc = {
+			write_op,
+			&context
+	};
+
+	return  work_in_time_slice(&flash_op_desc);
+}
+
+#endif /* CONFIG_SOC_FLASH_NRF5_RADIO_SYNC */
+
+static int erase_op(void *context)
+{
+	u32_t prev_nvmc_cfg = NRF_NVMC->CONFIG;
+	u32_t pg_size = NRF_FICR->CODEPAGESIZE;
+	struct erase_context *e_ctx = context;
+
+#if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
+	u32_t ticks_begin = 0;
+	u32_t ticks_diff;
+	u32_t i = 0;
+
+	if (e_ctx->enable_time_limit) {
+		ticks_begin = ticker_ticks_now_get();
+	}
+#endif /* CONFIG_SOC_FLASH_NRF5_RADIO_SYNC */
+
+	/* Erase uses a specific configuration register */
+	NRF_NVMC->CONFIG = NVMC_CONFIG_WEN_Een << NVMC_CONFIG_WEN_Pos;
+	nvmc_wait_ready();
+
+	do	{
+		NRF_NVMC->ERASEPAGE = e_ctx->addr;
+		nvmc_wait_ready();
+
+		e_ctx->size -= pg_size;
+		e_ctx->addr += pg_size;
+
+#if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
+		i++;
+
+		if (e_ctx->enable_time_limit) {
+
+			ticks_diff = ticker_ticks_now_get() - ticks_begin;
```

Comment:
You can remove the empty line after the if-statement. It's only needed after variable declarations if a block has those.